pdi-engine-api/api/src/main/java/org/pentaho/di/engine/api/remote/Execution.java

```diff
@@ -0,0 +1,36 @@
+package org.pentaho.di.engine.api.remote;
+
+import org.pentaho.di.engine.api.events.PDIEvent;
+import org.pentaho.di.engine.api.model.LogicalModelElement;
+
+import java.io.InputStream;
+import java.io.Serializable;
+
+/**
+ * Service representing a single transformation execution on a remote engine.
+ * <p>
+ * Created by hudak on 2/7/17.
+ */
+public interface Execution {
+  /**
+   * @return original request for this execution
+   */
+  ExecutionRequest getRequest();
+
+  /**
+   * Update a {@link LogicalModelElement} from this request.
+   * This will be routed to through the event pipe to the client.
+   *
+   * @param sourceId {@link LogicalModelElement#getId()}
+   * @param value    Serialized value of {@link PDIEvent#getData()}
+   */
+  void update( String sourceId, byte[] value );
```

Comment:
public interface Execution<T extends Serializable>
update( T payload )

Then the client impl can bind it to TunneledPayload ??